contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+                    sqrtPriceX96: 0
+                })
+            );
+        } else {
+            (, _wethAmountSupplied, _oethbAmountSupplied) = positionManager
+                .increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+
+        /**
+         * First check we are in expected tick range
+         *
+         * We revert even though price being equal to the lower tick would still
+         * count being within lower tick for the purpose of Sugar.estimateAmount calls
+         */
+        if (
+            _currentPrice <= sqrtRatioX96TickLower ||
+            _currentPrice >= sqrtRatioX96TickHigher
+        ) {
+            revert OutsideExpectedTickRange(getCurrentTradingTick());
+        }
+
+        /**
+         * If estimateAmount1 call fails it could be due to _currentPrice being really
+         * close to a tick and amount1 too big to compute.
+         *
+         * If token addresses were reversed estimateAmount0 would be required here
+         */
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOethAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(0), // no need to pass pool address when current price is specified
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOethAmount
+        );
+
+        if (
+            _wethSharePct >= allowedWethShareStart &&
```

Comment:
🔴 Note that until we set these `allowedWethShareStart` and `allowedWethShareEnd` they are zero. And because of rounding down, _wethSharePct could return a zero on non-zero ratio of coins. This could possibly trigger a massive min. Maybe. Not going to spend the time to test it out, but this basically defaults to worst case value.

Maybe we just throw a require above on this that allowedWethShareStart must be > 0.

Reply:
This could/would happen on a faulty deploy where `setAllowedPoolWethShareInterval` wouldn't be called since the minimum is [limited to 1%](https://github.com/OriginProtocol/origin-dollar/pull/2184/files#diff-bf8f6dd8d073ea49c4bcb1dd72c9980951407dbcb082474292a4f5f4696d4956R286) in 18 decimal format. Sanity check added here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/260c9db416bf46221191f80fec08531b16c14622